bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -89,138 +91,147 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.gson.GsonBuilder;
+
 /**
  * {@link ApiBridgeHandler} is the handler for a Netatmo API and connects it to the framework.
  *
  * @author Gaël L'hopital - Initial contribution
- *
+ * @author Jacob Laursen - Refactored to use standard OAuth2 implementation
  */
 @NonNullByDefault
 public class ApiBridgeHandler extends BaseBridgeHandler {
     private static final int TIMEOUT_S = 20;
-    private static final String REFRESH_TOKEN = "refreshToken";
 
     private final Logger logger = LoggerFactory.getLogger(ApiBridgeHandler.class);
-    private final AuthenticationApi connectApi = new AuthenticationApi(this, scheduler);
+    private final AuthenticationApi connectApi = new AuthenticationApi(this);
     private final Map<Class<? extends RestManager>, RestManager> managers = new HashMap<>();
     private final Deque<LocalDateTime> requestsTimestamps = new ArrayDeque<>(200);
     private final BindingConfiguration bindingConf;
     private final HttpClient httpClient;
+    private final OAuthFactory oAuthFactory;
     private final NADeserializer deserializer;
     private final HttpService httpService;
     private final ChannelUID requestCountChannelUID;
-    private final Path tokenFile;
 
+    private @Nullable OAuthClientService oAuthClientService;
     private Optional<ScheduledFuture<?>> connectJob = Optional.empty();
     private Optional<WebhookServlet> webHookServlet = Optional.empty();
     private Optional<GrantServlet> grantServlet = Optional.empty();
 
     public ApiBridgeHandler(Bridge bridge, HttpClient httpClient, NADeserializer deserializer,
-            BindingConfiguration configuration, HttpService httpService) {
+            BindingConfiguration configuration, HttpService httpService, OAuthFactory oAuthFactory) {
         super(bridge);
         this.bindingConf = configuration;
         this.httpClient = httpClient;
         this.deserializer = deserializer;
         this.httpService = httpService;
-        this.requestCountChannelUID = new ChannelUID(thing.getUID(), GROUP_MONITORING, CHANNEL_REQUEST_COUNT);
+        this.oAuthFactory = oAuthFactory;
 
-        Path homeFolder = Paths.get(OpenHAB.getUserDataFolder(), BINDING_ID);
-        if (Files.notExists(homeFolder)) {
-            try {
-                Files.createDirectory(homeFolder);
-            } catch (IOException e) {
-                logger.warn("Unable to create {} folder : {}", homeFolder.toString(), e.getMessage());
-            }
-        }
-        tokenFile = homeFolder.resolve(REFRESH_TOKEN + "_" + thing.getUID().toString().replace(":", "_"));
+        requestCountChannelUID = new ChannelUID(thing.getUID(), GROUP_MONITORING, CHANNEL_REQUEST_COUNT);
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Netatmo API bridge handler.");
         updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Please move this line below, at least after your 2 checks of valid client id and client secret. Or just before your call to scheduler.execute.
This will avoid a doubled thing state update.